### PR TITLE
strcmp optimization for riscv

### PR DESCRIPTION
**Pull request by M-Moawad**
Optimize strcmp by calling orc.b (OR-Combine) instruction when Zbb extension is supported instead of 4 instructions to detect if zero byte is exist in word

### CODE REVIEW

File: newlib/libc/machine/riscv/strcmp.S [hunk 1]
```diff
@@ -46,12 +46,14 @@ strcmp:
   .macro check_one_word i n
     REG_L a2, \i*SZREG(a0)
     REG_L a3, \i*SZREG(a1)
-
+#ifdef __riscv_zbb
+    orc.b t2, t0
```
Comment (keith-packard) [inline newlib/libc/machine/riscv/strcmp.S]: Reading the adjacent code, it seems like this should be
```
    orc.b t0, a2
```
that is, t0 should contain 0xff in each non-zero byte of a2.

**Comment by keith-packard** [comment]
I also hacked up one of the CI configurations to enable Zbb for testing:
```
diff --git a/scripts/cross-rv32imac_zicsr.txt b/scripts/cross-rv32imac_zicsr.txt
index ef1b2751ee..6e12ac283d 100644
--- a/scripts/cross-rv32imac_zicsr.txt
+++ b/scripts/cross-rv32imac_zicsr.txt
@@ -14,8 +14,8 @@ cpu = 'riscv32'
 endian = 'little'
 
 [properties]
-c_args = ['-msave-restore', '-fshort-enums', '-march=rv32imac_zicsr', '-mabi=ilp32']
-c_link_args = ['-msave-restore', '-fshort-enums', '-march=rv32imac', '-mabi=ilp32']
+c_args = ['-msave-restore', '-fshort-enums', '-march=rv32imac_zicsr_zbb', '-mabi=ilp32']
+c_link_args = ['-msave-restore', '-fshort-enums', '-march=rv32imac_zicsr_zbb', '-mabi=ilp32']
 skip_sanity_check = true
 default_flash_addr = '0x80000000'
 default_flash_size = '0x00200000'
```

**Comment by keith-packard** [comment]
It also looks like the mask value stored in 'a5' is not needed any more, so we can remove initializing that register and the related 'mask' value at the bottom of the file.

File: newlib/libc/machine/riscv/strcmp.S (hunk 1 above)
Comment (M-Moawad) [reply newlib/libc/machine/riscv/strcmp.S]: Edited on the same commit

**Comment by M-Moawad** [comment]
> It also looks like the mask value stored in 'a5' is not needed any more, so we can remove initializing that register and the related 'mask' value at the bottom of the file.

Check the current change, I moved the value assigned to `a5` to be used when `zbb` extension is disabled, also the related mask at the bottom of the file will be stored only in this case.

**Comment by keith-packard** [comment]
I updated your last commit to add `_zbb` to `do-rv32imac-configure` which is what CI uses to run that test.

**Pull request merged**